File: src/api/scan.ts
import { invoke } from "@tauri-apps/api/core";
import type { BookWithTags } from "./types";

export interface ScanReport {
  total: number;
  added: number;
  updated: number;
  errors: string[];
}

export interface ScanProgress {
  current: number;
  total: number;
  current_file: string;
}

export interface SearchFilter {
  keyword?: string;
  tag_ids?: number[];
  is_read?: number;
  min_rating?: number;
  sort_by?: string;
  sort_order?: string;
}

/** 扫描目录 */
export async function scanDirectory(root: string): Promise<ScanReport> {
  return invoke("scan_directory", { root });
}

/** 搜索图书 */
export async function searchBooks(filter: SearchFilter): Promise<BookWithTags[]> {
  return invoke("search_books", { filter });
}

File: src-tauri/src/commands/scan.rs
use crate::models::*;
use crate::services::{scanner, search};
use tauri::command;
use tauri::AppHandle;

/// 扫描目录
#[command]
pub async fn scan_directory(app_handle: AppHandle, root: String) -> Result<ScanReport, String> {
    scanner::scan_directory(app_handle, &root)
}

/// 搜索图书
#[command]
pub fn search_books(filter: SearchFilter) -> Result<Vec<BookWithTags>, String> {
    search::search_books(&filter)
}
